Remove web chimera dependency

guidupuy opened the pull request:
Since the developer can choose how they want to supply their `wcjs` instance (using `webchimera.js`, or with `wcjs-prebuilt`), removing `webchimera.js` from the dependencies and simply making the `wcjs` parameter mandatory in the API seems quite sensible...

An argument for this is that for developers planning on using `wcjs-prebuilt`, having `npm` install `webchimera.js` makes little sense...

guidupuy committed:
Remove web chimera dependency

Since the developer can choose what way to create her wcjs instance
(web chimera.js or wcjs-prebuilt), we shouldn’t include webchimera as a
dependency and simply make the wcjs parameter mandatory in the API.

guidupuy committed:
Bump up version

This new version introduces breaking changes to the API - increment
major version to comply with semver.

RSATom commented:
You could just use https://www.npmjs.com/package/webgl-video-renderer. It's almost the same, just a little bit simpler, and already with ability specify external wcjs object.

guidupuy commented:
I could... But I still think the  change I suggested makes sense from an API design standpoint ;-)

Forcing a user to download a dependency that's not necessarily needed doesn't seem like a good idea.

RSATom commented:
The main problem is if we get rid of wcjs dependancy, it will break existing software which already use wcjs-renderer. That's why I've forked it and changed name to `webgl-video-renderer`.

guidupuy commented:
Breaking changes happen all the time in software development, and not implementing sensible changes just because it breaks existing software is a rather weak argument, IMHO. 

Don't get me wrong: breaking existing software _is_ a problem, but the solution isn't to stop improving your code just to maintain backwards compatibility. A much better solution is:
- On the software vendor side, to use semantic [semantic versioning](http://semver.org/)'s `MAJOR` version increments when breaking changes are introduced
- On the developer side, to use the `^` symbol in your `package.json` dependencies to fix the `MAJOR` version of the package you're using and therefore avoid breaking changes

In this specific case, people using the old API would use `"wcjs-renderer": "^0.1.10"` in their `package.json` dependencies and be done with it... 

Using `latest` when specifying a dependency is considered bad practice at this point anyway, and `npm` now uses the `^` symbol by default when using `npm install <package> --save`. Anyone who's installed `wcjs-renderer` this way (which I think can be considered the standard way) will therefore have `"wcjs-renderer": "^0.1.10"` in their dependencies and see no breaking change when running the `npm update` command.

guidupuy committed:
New: The renderer shouldn't create its own vlc player

Making the vlc player an input parameter means the developer has more
flexibility on reusing an existing player vs. instantiating a new one.

guidupuy committed:
Fix: The 'bind' function shouldn't return anything

RSATom commented:
Why just not use `webgl-video-renderer`? It was forked exactly to get rid of direct wcjs dependency.

RSATom commented:
@luigiplr, @Ivshti I don't mind if you will merge this pull request...

guidupuy commented:
It's just a suggestion, really. Feel free to do as you please. I think it's better without the dependency, but as far as I'm concerned I can always use my own fork...

luigiplr commented:
Hmmm, modularity is a very attractive feature indeed.

Just one thing I'm not sure about; Why did the version get bumped to `2.0.0` from `0.1.10` ?

guidupuy commented:
Following my first suggestion I did some more breaking change to the API and
subsequently bumped the version one more time. But if you're going to merge the pull request then version can stay at 1.0.0... I'll change that.

Le dimanche 1 mai 2016, Luigi Poole notifications@github.com a écrit :

> Hmmm, modularity is a very attractive feature indeed.
> 
> Just one thing I'm not sure about; Why did the version get bumped to 2.0.0
> from 0.1.10 ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/Magics-Group/wcjs-renderer/pull/13#issuecomment-216064208

guidupuy committed:
Reverted version to 1.0.0

luigiplr commented:
Looking good!

Thanks @guidupuy, It's much appreciated.

jaruba commented:
If you publish this, all the `wcjs-player` users will come knocking on my door..

jaruba commented:
With axes nonetheless..

![yekurll](https://cloud.githubusercontent.com/assets/1777923/15091655/788d77e2-1458-11e6-8ada-ff6634eec009.gif)

luigiplr commented:
The semvar has been updated appropriately to reflect a major api change.

jaruba commented:
@luigiplr 
![yeah_eddie_murphy](https://cloud.githubusercontent.com/assets/1777923/15091710/b6deb5be-1459-11e6-8d6c-d2ed7c64832c.gif)

RSATom commented:
@jaruba, you have a solution - switch to `webgl-video-renderer` :)

jaruba commented:
@RSATom nope, this PR actually makes sense, there was already a PR for `wcjs-player` that allows people to push whatever `wcjs` object they have to the player, be it `wcjs-prebuilt` or a self built version.

I just need to update my `README.md` for it :)

jaruba commented:
The only thing I'm actually scared of is making the installation even more complicated then it already is for new comers and explaining to all the people that are building `wcjs` atm that things have changed in the installation process.

But that's not a serious reason to create drama over it. :)

jaruba commented:
I published this change today (including all the other PRs from @guidupuy ) after applying some fixes to the code. Thanks for all your hard work @guidupuy , I'm loving the `wcjs-prebuilt` changes. ;)

[wcjs-player](https://github.com/jaruba/wcjs-player/) and it's two demo app repos have also been updated to the new `wcjs-renderer` JS API and Docs changed accordingly for this major change.

fleex commented:
Great! Thanks!
